### PR TITLE
add support for Transfer Server endpoint and alphabetize variables

### PR DESCRIPTION
**Pull request by marwinbaumannsbp**

### CODE REVIEW

File: endpoints.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  service = "transfer.server"
+}
+
+resource "aws_vpc_endpoint" "transferserver" {
+  count               = var.transferserver_endpoint != null ? 1 : 0
+  private_dns_enabled = var.transferserver_endpoint.private_dns_enabled
+  security_group_ids  = var.transferserver_endpoint.security_group_ids
+  service_name        = data.aws_vpc_endpoint_service.transferserver[0].service_name
+  subnet_ids          = var.transferserver_endpoint.subnet_ids
+  tags                = merge(var.tags, { "Name" = "${var.stack}-transferserver_endpoint" })
```
Comment (svanharmelen) [inline endpoints.tf]: I suggest to use either `-` or `_` in the name, but not both.

File: endpoints.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+data "aws_vpc_endpoint_service" "transferserver" {
```
Comment (shoekstra) [inline endpoints.tf]: maybe nitpicking, but can instances of `transferserver` be changed to `transfer_server` please?

File: endpoints.tf (hunk 1 above)
Comment (shoekstra) [reply endpoints.tf]: Adding to this, can we drop the `-endpoint` suffix please? We know it's an endpoint by the type of resource.  

And following on from previous comment this should look like `"${var.stack}-transfer-server"`, I'm not a fan of concatenating words when already using kebab- or snake-case in a string.

Comment (marwinbaumannsbp) [reply endpoints.tf]: Ok

File: endpoints.tf (hunk 2 above)
Comment (marwinbaumannsbp) [reply endpoints.tf]: I thought it was specified as one word in the CLI but its indeed seperated by a "." indicating 2 words. Changed this to transfer_server.

**Pull request merged**